# joint mask array
bool[] mask